### PR TITLE
TB-1884 Replace Access Denied with a redirect

### PR DESCRIPTION
**Pull request by Kingdutch**
Instead of giving the user an error that leaves them stuck when they try
to start a course for the second time, simply redirect them to the first
material in the course. From there they can find their way around.

The most likely scenario where they visit the link twice will also be
when they are still at the first part.

### CODE REVIEW

File: src/Controller/CoursesController.php [hunk 1]
```diff
@@ -77,6 +77,22 @@ public static function access(AccountInterface $account) {
   public function startSection(GroupInterface $group, NodeInterface $node) {
     // Get first material.
     $material = $node->get('field_course_section_content')->get(0)->entity;
+    $material_redirect = $this->redirect('entity.node.canonical', [
+      'node' => $material->id(),
+    ]);
+
+    /** @var \Drupal\Core\Entity\EntityStorageInterface $course_enrollment_storage */
+    $course_enrollment_storage = \Drupal::entityTypeManager()->getStorage('course_enrollment');
```
Comment (OH17L) [inline src/Controller/CoursesController.php]: Use injection.

File: src/Controller/CoursesController.php [hunk 2]
```diff
@@ -77,6 +77,22 @@ public static function access(AccountInterface $account) {
...
+    $entities = $course_enrollment_storage->loadByProperties([
+      'uid' => \Drupal::currentUser()->id(),
```
Comment (OH17L) [inline src/Controller/CoursesController.php]: Use injection.

**Comment by Kingdutch** [comment]
Hey @OH17L ,

Given that dependency injection wasn't used anywhere in this yet I stuck with doing things the same as in the rest of the class. This keeps the change small.

I think we should refactor the dependency injection in a follow-up story as more work will be needed and it could break things differently.

**Comment by OH17L** [comment]
@Kingdutch, services which using in this class already defined in parent class so you should just change **\Drupal::entityTypeManager()** to **$this->entityTypeManager**. These changes have not needed a lot of time but we can do it later.

**Comment by Kingdutch** [comment]
Good point @OH17L : )

I'll do it in the next PR!

**Pull request merged**